Update Npgsql EF Core integration to latest versions

eerhardt opened the pull request:
This will allow us to ship `Aspire.Npgsql.EntityFrameworkCore.PostgreSQL` as stable `9.0.0`. This should be backported to release/9.0.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/6721)

eerhardt committed:
Update Npgsql EF Core integration to latest versions

joperezr reviewed (approved):
Can we also remove the following as part of this change:

https://github.com/dotnet/aspire/blob/95eec9c312612e2e20e2663eb83aeccfe5e495d3/src/Components/Aspire.Npgsql.EntityFrameworkCore.PostgreSQL/Aspire.Npgsql.EntityFrameworkCore.PostgreSQL.csproj#L9-L10

eerhardt committed:
Remove SuppressFinalPackageVersion so it ships stable

Also update Npgsql DependencyInjection and OpenTelemetry to latest version, since these are referenced by the EF package.

eerhardt commented:
> Can we also remove the following as part of this change:
> 
> https://github.com/dotnet/aspire/blob/95eec9c312612e2e20e2663eb83aeccfe5e495d3/src/Components/Aspire.Npgsql.EntityFrameworkCore.PostgreSQL/Aspire.Npgsql.EntityFrameworkCore.PostgreSQL.csproj#L9-L10

Good call. Removed.

eerhardt commented:
/backport to release/9.0

eerhardt committed:
Fix package validation for Npgsql EF
